src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java

```diff
@@ -121,51 +124,117 @@ else if( clazz.equals( Long.class )) {
 
 				String query =
 					"select *" +
-					" from (" + definition.virtualDataTable + ") as data" +
-					" left join (" + definition.virtualInstanceTable + ") instance" +
+					" from (" + definition.dataSource.virtualDataTable + ") as data" +
+					" left join (" + definition.dataSource.virtualInstanceTable + ") instance" +
 					" on instance.id = data.instance_id" +
 					" where " + selectionSql +
 					" and data.instance_id = " + Long.toString( instanceId) +
 					" order by " + orderClause +
 					";" ;
 
-				// TODO, useful to monitor integration tests - remove later,
+				// TODO logger
 				System.out.println( "instanceId is " + instanceId + " " + query);
 
-
-				populateValues( query, definition.dimensions, definition.encoders );
+				populateValues( query, definition.dimensions, definition.variables );
 
 				NetcdfFileWriteable writer = createWritable.create();
 
+				// Write the global attributes
+				for( Attribute attribute: definition.globalAttributes )
```

Comment:
Agree that it can and should be factored out. I'd prefer to hold off though, until the other requirements are known before making the change.  For example, I suspect it's quite likely that we'll want to support variable attribute sql lookup and this will influence the final shape of the code.